src/gcode-machine-control.cc

```diff
@@ -530,13 +529,21 @@ void GCodeMachineControl::Impl::set_output_flags(HardwareMapping::LogicOutput ou
 }
 
 void GCodeMachineControl::Impl::get_current_position() {
-  if (planning_buffer_.size() > 0) {
-    const struct AxisTarget *current = planning_buffer_[0];
-    const int *mpos = current->position_steps;
-    const float x = 1.0f * mpos[AXIS_X] / cfg_.steps_per_mm[AXIS_X];
-    const float y = 1.0f * mpos[AXIS_Y] / cfg_.steps_per_mm[AXIS_Y];
-    const float z = 1.0f * mpos[AXIS_Z] / cfg_.steps_per_mm[AXIS_Z];
-    const float e = 1.0f * mpos[AXIS_E] / cfg_.steps_per_mm[AXIS_E];
+    int i, motor;
+    const MotorCoordSystem
+        &motor_position_steps_ = motor_ops_->GetCurrentMotorPos();
```

Comment:
The underscore should only be used for instance variables, not local variables.

You are assigning this to a const reference here. This will work the way c++ guarantees temporary references to exist in the context, but it is probably not what you meant (but instead, meant to copy the coordinates).

Speaking of which, is it not possible hat GetCurrentMotorPos() returns a reference to the current position ? I guess I find out later in the review.